scripts/SREP/describe-nodes/script.sh

```diff
@@ -0,0 +1,189 @@
+# echo "ARG_ARRAY: ${ARG_ARRAY}"
+while [[ ${#ARG_ARRAY[@]} -gt 0 ]]
+do
+  # echo "in while"
+  case ${ARG_ARRAY[0]} in
+    -h|--help)
+      error-out
+      ;;
+    -a|--all)
+      all=true
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -m|--master)
+      master=true
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -i|--infra)
+      infra=true
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -w|--worker)
+      worker=true
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -d|--debug)
+      set -x  # Print expanded commands with arguments
+      ARG_ARRAY=("${ARG_ARRAY[@]:1}")  # shift past argument
+      ;;
+    -n|--node|--nodes)
+      if [ ! -z "$nodes" ]
+      then
+        error-out "Only 1 '--nodes' argument can be provided"
+      fi
+      # Check for unset or empty arguments and values that look like a flag/switch
+      if [ ${#ARG_ARRAY[@]} -le 1 ] \
+        || [ -z ${ARG_ARRAY[1]} ] \
+        || [[ ${ARG_ARRAY[1]} =~ ^-.* ]]
+      then
+        error-out "The --nodes requires a list of nodes"
+      fi
+      # Check for ilegal characters in the list of nodes
+      badchars="$(echo ${ARG_ARRAY[1]} | tr -d '[:alnum:]' | tr -d '-' | tr -d '.' | tr -d ',')"
+      if [ ! -z ${badchars} ]
+      then  
+        error-out "'--nodes' can only contain '.|-|,|<alphanumric>' characters.  These characters '${badchars}' are ileagal"
+      fi
+      nodes=${ARG_ARRAY[1]}
+      ARG_ARRAY=("${ARG_ARRAY[@]:2}")  # shift past argument and value
+      ;;
+    -l|--selector)
+      if [ ! -z "$selector" ]
+      then
+        error-out "Only 1 '--selector' argument can be provided"
+      fi
+      # Check for unset or empty arguments and values that look like a flag/switch
+      if [ ${#ARG_ARRAY[@]} -le 1 ] \
+        || [ -z ${ARG_ARRAY[1]} ] \
+        || [[ ${ARG_ARRAY[1]} =~ ^-.* ]]
+      then
+        error-out "'--selector' must contain a node selector"
+      fi
+      badchars="$(echo ${ARG_ARRAY[1]} | tr -d '[:alnum:]' | tr -d '-' | tr -d '.'  | tr -d '/'  \
+          | tr -d ',' | tr -d ':' | tr -d ' ' | tr -d '(' | tr -d ')' | tr -d '!' | tr -d '=')"
+      if [ ! -z ${badchars} ]
+      then
+        error-out "--selector ilegal caharcters in selector.  These characters '${badchars}' are ileagal"
+      fi
+      selector=${ARG_ARRAY[1]}
+      ARG_ARRAY=("${ARG_ARRAY[@]:2}")  # shift past argument and value
+      ;;
+    -*|--*)
+      error-out "Unknown option ${ARG_ARRAY[0]}"
+      ;;
+    *)
+      error-out "Unknown argument"
+      ;;
+  esac
+done
+
+# If --nodes is supplied use that and ignore all other arguments
+# else if --selector is supplied use that ignore the otheres arguments
+# else if --all is supplied use that ignore the otheres arguments
+# else if any of the --master or --infra or --worker arguments are supplied use them all
+if [ ! -z "$nodes" ]
+then
+  
```

Comment:
No actual command between the `then` and the `elif` is invalid bash - shellcheck (in the PR check) is currently picking this up as a problem. If nothing is meant to happen here, that whole if test can be removed.

Note that as soon as that's fixed, shellcheck is going to give present you with a handful of other issues which need to be addressed to pass linting checks. I'll cover a few of them here, but there's a bunch, so it'll probably be easier to review the PR check output (or run `shellcheck` manually over your script) to fix them all.

Reply:
Fixed and re- tested

`If nothing is meant to happen here, that whole if test can be removed.`  Unfortunately it is required to enforce the precedence order since I only allow one mode of the the command to run...  See the comment at the start of the iff statement.

But it is poor programming, I have fliped the test and moved the ELIF to the true section of the test.

In the previous commit there was a break, but I realised this was also invalid so I had removed it... I was in 2 minds weather to drop a non impacting cmd (equivalent to python `skip`) in there (like `echo -n`) or re-write,..... My 2 minds had walked away from the problem with out solving it.